File: JS/stackful.js
function* profundidade() {
    console.log("Antes de pausar");
    yield;
    console.log("Depois de retomar");
}
    
function* gerador() {
    yield* profundidade();
}
    
const g = gerador();
g.next(); // "Antes de pausar"
g.next(); // "Depois de retomar"

File: Lua/stackful.lua
function profundidade()
    print("Antes de pausar")
    coroutine.yield()
    print("Depois de retomar")
end

function corrotina()
    profundidade()
end

co = coroutine.create(corrotina)
coroutine.resume(co)
coroutine.resume(co)